core-bundle/src/Filesystem/Dbafs/Dbafs.php

```diff
@@ -788,18 +790,18 @@ private function getFilesystemPaths(array $searchPaths, array $parentDirectories
             }
 
             if (null === ($isDir = $analyzedPaths[$searchPath] ?? null)) {
-                if (!$shallow && $this->filesystem->fileExists($searchPath, VirtualFilesystemInterface::BYPASS_DBAFS)) {
-                    // Yield file
-                    yield $searchPath => self::RESOURCE_FILE;
-
-                    continue;
-                }
-
                 // Analyze parent path
                 $analyzedPaths = array_merge($analyzedPaths, $analyzeDirectory(Path::getDirectory($searchPath)));
                 $isDir = $analyzedPaths[$searchPath] ??= false;
             }
 
+            if (!$isDir && !$shallow && $this->filesystem->fileExists($searchPath, VirtualFilesystemInterface::BYPASS_DBAFS)) {
+                // Yield existing file
+                yield $searchPath => self::RESOURCE_FILE;
+
+                continue;
+            }
+
             if ($isDir) {
                 yield from $traverseRecursively($searchPath, $shallow);
 
```

Comment:
```suggestion
            if ($isDir) {
                yield from $traverseRecursively($searchPath, $shallow);

                continue;
            }
            
            if (!$shallow && $this->filesystem->fileExists($searchPath, VirtualFilesystemInterface::BYPASS_DBAFS)) {
                // Yield existing file
                yield $searchPath => self::RESOURCE_FILE;

```

The `!$isDir` check would not be needed if we move the code after the `if ($isDir)` early return I think?

Reply:
True, that. Thanks for the suggestion. :heart: